File: WorkersApplication-main/app/src/main/java/com/example/workersapplication/Adapters/WorkersListAdapter.java
package com.example.workersapplication.Adapters;

import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.workersapplication.Models.Workers;
import com.example.workersapplication.R;
import com.example.workersapplication.WorkersProfile;

import java.util.ArrayList;

public class WorkersListAdapter extends RecyclerView.Adapter<WorkersListAdapter.MyViewHolder>{

    ArrayList<Workers> list;

    public WorkersListAdapter(ArrayList<Workers> list) {
        this.list = list;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.model_workers_list,parent,false);
        return new MyViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
       Workers workers = list.get(position);
       holder.worker_name.setText(workers.getWorker_name());
       holder.location.setText(workers.getLocation());
       holder.worker_pic.setImageResource(workers.getWorker_image());
       holder.rating.setText(workers.getRating());
       holder.salary.setText(workers.getSalary());

        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(holder.itemView.getContext(), WorkersProfile.class);
                holder.itemView.getContext().startActivity(intent);
            }
        });
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    class MyViewHolder extends RecyclerView.ViewHolder {
        TextView worker_name,location,salary,rating;
        ImageView worker_pic,bookmark_btn;
        int bookmark_select = 0;
        public MyViewHolder(@NonNull View itemView) {
            super(itemView);
            worker_name = itemView.findViewById(R.id.worker_name);
            location = itemView.findViewById(R.id.worker_location);
            worker_pic = itemView.findViewById(R.id.worker_image);
            bookmark_btn = itemView.findViewById(R.id.bookmark_btn);
            salary = itemView.findViewById(R.id.salary);
            rating = itemView.findViewById(R.id.rating);
            worker_pic.setClipToOutline(true);

            bookmark_btn.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    if (bookmark_select == 0){
                        bookmark_select = 1;
                        bookmark_btn.setImageResource(R.drawable.ic_baseline_bookmark_24);
                    }
                    else {
                        bookmark_select = 0;
                        bookmark_btn.setImageResource(R.drawable.ic_baseline_bookmark_border_24);
                    }
                }
            });

        }
    }
}

File: WorkersApplication-main/app/src/main/java/com/example/workersapplication/fragments/list/WorkersListAdapter.kt
package com.example.workersapplication.fragments.list

import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.workersapplication.KotSettingsActivity
import com.example.workersapplication.R
import com.example.workersapplication.WorkersProfile
import kotlinx.android.synthetic.main.activity_main.*
import com.example.workersapplication.database.workers.Workers
import kotlinx.android.synthetic.main.model_workers_list.view.*

class WorkersListAdapter: RecyclerView.Adapter<WorkersListAdapter.MyViewHolder>() {

    private var workersList = emptyList<Workers>()

    class MyViewHolder(itemView: View): RecyclerView.ViewHolder(itemView){


    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        return MyViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.model_workers_list, parent, false))
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val currentItem = workersList[position]
        holder.itemView.worker_name.text = currentItem.fName + " "+ currentItem.lName
        holder.itemView.worker_location.text = currentItem.locality + " ,"+ currentItem.city
        holder.itemView.workers_age.text = currentItem.age.toString()
        holder.itemView.rating.text = currentItem.rating.toString()
        holder.itemView.families_cared.text = "Cared for "+ currentItem.families_cared.toString() +" families"
        holder.itemView.experience.text = currentItem.experience.toString() +" years paid experience"
        holder.itemView.salary.text = "₹ "+currentItem.price.toString() +"/Mon"

        holder.itemView.setOnClickListener {
            val intent = Intent(holder.itemView.context, WorkersProfile::class.java)
            intent.putExtra("wname",currentItem.fName + " "+ currentItem.lName)
            intent.putExtra("waddress",currentItem.locality + " ,"+ currentItem.city)
            intent.putExtra("wage",currentItem.age.toString())
            intent.putExtra("wrating",currentItem.rating.toString())
            intent.putExtra("wprice",currentItem.price.toString())
            intent.putExtra("wexp",currentItem.experience.toString())
            //intent.putExtra("wrating",currentItem.rating.toString())
            holder.itemView.context.startActivity(intent)
        }

    }

    override fun getItemCount(): Int {
        return workersList.size
    }

    fun setData(worker: List<Workers>){
        this.workersList = worker
        notifyDataSetChanged()
    }
}